python/lsst/validate/drp/checkAstrometryPhotometry.py

```diff
@@ -0,0 +1,469 @@
+    """
+    ra_avg, dec_avg = averageRaDec(cat)
+    ra, dec = cat.get('coord_ra'), cat.get('coord_dec')
+    # Approximating that the cos(dec) term is roughly the same
+    #   for all observations of this object.
+    ra_var = np.var(ra) * np.cos(dec_avg)**2
+    dec_var = np.var(dec)
+    pos_rms = np.sqrt(ra_var + dec_var)  # radians
+    pos_rms = afwGeom.radToMas(pos_rms)  # milliarcsec
+
+    return pos_rms
+
+
+def loadAndMatchData(repo, visitDataIds,
+                     matchRadius=afwGeom.Angle(1, afwGeom.arcseconds)):
+    """Load data from specific visit.  Match with reference.
+
+    @param repo  The repository.  This is generally the directory on disk
+                    that contains the repository and mapper.
+    @param visitDataIds   Butler Data ID of Image catalogs to compare to reference.
+           The actual pixel image is also needed for now for the photometric calibration.
+           List.
+
+    @param matchRadius    Radius for matching.  afwGeom.Angle().
+
+    Return a afw.table.GroupView object of matched catalog.
+    """
+
+    # Following 
+    # https://github.com/lsst/afw/blob/tickets/DM-3896/examples/repeatability.ipynb
+    butler = dafPersist.Butler(repo)
+    dataset = 'src'
+
+    dataRefs = [dafPersist.ButlerDataRef(butler, vId) for vId in visitDataIds]
+
+    # retrieve the schema of the source catalog and extend it in order to add a field to record the ccd number
+    ccdKeyName = getCcdKeyName(visitDataIds[0])
+
+    schema = butler.get(dataset + "_schema", immediate=True).schema
+    mapper = SchemaMapper(schema)
+    mapper.addMinimalSchema(schema)
+#    mapper.addOutputField(Field[int](ccdKeyName, "CCD number"))
+    mapper.addOutputField(Field[float]('base_PsfFlux_mag', "PSF magnitude"))
+    mapper.addOutputField(Field[float]('base_PsfFlux_magerr', "PSF magnitude uncertainty"))
+    newSchema = mapper.getOutputSchema()
+
+    # Create an object that can match multiple catalogs with the same schema
+    mmatch = MultiMatch(newSchema, 
+                        dataIdFormat = {'visit': int, ccdKeyName: int},
+                        radius=matchRadius,
+                        RecordClass=SimpleRecord)
+
+    if False:
+        # Create visit catalogs by merging those from constituent CCDs. We also convert from BaseCatalog to SimpleCatalog, but in the future we'll probably want to have the source transformation tasks generate SimpleCatalogs (or SourceCatalogs) directly.
+        byVisit = {}
+        for dataRef in dataRefs:
+            catalog = byVisit.setdefault(dataRef.dataId["visit"], SimpleRecord.Catalog(schema))
+            catalog.extend(dataRef.get(dataset, immediate=True), deep=True)
+
+        # Loop over visits, adding them to the match.
+        for visit, catalog in byVisit.iteritems():
+            mmatch.add(catalog, dict(visit=visit))
+
+    # create the new extented source catalog
+    srcVis = SourceCatalog(newSchema)
+
+    for vId in visitDataIds:
```

Comment:
Yes, that's exactly right.

TransformTask sounds promising, but I fear that I can't quickly convert this to use TransformTask without some better documentation or example of its use.

Can you point me to what the equivalent way to write this would be using TransformTask?

Why is it a "Task"?  As far as I can tell, I get a source catalog back as the output of TransformTask.run().    I'm reading through

https://github.com/lsst/pipe_tasks/blob/5e752b3323af3387d8b07baf80128070c86db387/tests/testTransform.py#L169

and trying to figure out how this might work.